### PR TITLE
[vcpkg] Onboard Linux to VMSS, open 'git' port, and switch back to Azure Spot

### PR DESCRIPTION
**Pull request by BillyONeal**
* Adds scripts to generate scale sets for testing Linux.
    * Note workaround for https://github.com/microsoft/azure-pipelines-agent/pull/2929
* Switches Windows validation to 'Spot' VMs.
* Opens the git port 9418.
* Removes provisioning of the no longer used 'logs' file share.
* Changes Azure region to 'westus2', which is cheaper.
* Adds +x to all the scripts in scripts/azure-pipelines.
* Use 'xml-results' for all platforms instead of 'raw xml results' on Windows.

### CODE REVIEW

File: scripts/azure-pipelines/macos/azure-pipelines.yml [hunk 1]
```diff
@@ -63,7 +66,7 @@ jobs:
       arguments: '-baselineFile $(System.DefaultWorkingDirectory)/scripts/ci.baseline.txt -logDir "$(System.ArtifactsDirectory)/xml-results" -outputDir "$(Build.ArtifactStagingDirectory)" -failurelogDir ~/Data/archives/fail -triplets "x64-osx" -errorOnRegression'
 
   - task: PublishBuildArtifacts@1
-    displayName: 'Publish Artifact: x64-osx port build failure logs'
+    displayName: 'Publish Artifact: x64_osx port build failure logs'
```
Comment (BillyONeal) [inline scripts/azure-pipelines/macos/azure-pipelines.yml:69]: I left this as dash in the other ones so I should here too.

File: scripts/azure-pipelines/windows/create-vmss.ps1 [hunk 2]
```diff
@@ -282,10 +293,8 @@ $StorageContext = New-AzStorageContext `
   -StorageAccountName $StorageAccountName `
   -StorageAccountKey $StorageAccountKey
 
-$ArchivesFiles = New-AzStorageShare -Name 'archives' -Context $StorageContext
+New-AzStorageShare -Name 'archives' -Context $StorageContext
 Set-AzStorageShareQuota -ShareName 'archives' -Context $StorageContext -Quota 5120
```
Comment (BillyONeal) [inline scripts/azure-pipelines/windows/create-vmss.ps1:297]: Given that we're going to recycle this somewhat more frequently, and given that each platform now gets their own, is 5TiB still an appropriate quota limit?

Comment (cenit) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: llvm and qt are big, considering debug+release configs on windows + pdb symbol files, 5 TiB might still be reasonable when you want to rebuild the full port tree (now we run into full-disk problems)

Comment (BillyONeal) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: @cenit A full tree rebuild only consumes about 70GiB of 'archives' space. If there are full disk problems they aren't coming from this quota setting.

Comment (cbezault) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: Agreed, it's the downloads + buildtree + packages + install directories that get massive.

Comment (cbezault) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: To answer Billy's original question, I think 5TiB could probably be dropped for other platforms but I think is appropriate for windows. 6 configurations * 70 GiB * 10 full rebuilds (which has happened in the span of a week before) gets us pretty close to 5TiB

Comment (BillyONeal) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: @cbezault That 70GiB is for all 6 configurations. (I did a full rebuild and asked Azure what our space consumption on the share was afterwards)

Comment (cbezault) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: Okay so 1TiB is sufficient, 2TiB is a guarantee

**Review by cbezault — APPROVED** [review]

File: ports/openblas/portfile.cmake [hunk 3]
```diff
@@ -3,13 +3,14 @@ include(vcpkg_common_functions)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO xianyi/OpenBLAS
-    REF 5f36f18148603facb6c3540e673610d6b24cbfbb # v0.3.7
-    SHA512 c1abcfd6fbf6ae3efa28053523ffb3d72aefaa279594347bfd9910c40f25778421b8d1d2d4dc6c934c9def40642942bb74fe76e06c2d96b1898ea61204a95540
+    REF v0.3.9
```
Comment (strega-nil) [inline ports/openblas/portfile.cmake:6]: This should probably use a git hash

File: ports/qt-advanced-docking-system/skip_license_install.patch [hunk 4]
```diff
@@ -0,0 +1,45 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index a2cb47d..57a1cf5 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -70,7 +70,7 @@ if("${CMAKE_SIZEOF_VOID_P}" STREQUAL "4")
+ else()
+     set(ads_PlatformDir "x64")
+ endif()
+-if(NOT BUILD_SHARED_LIBS)
++if(NOT BUILD_SHARED_LIBS)
```
Comment (strega-nil) [inline ports/qt-advanced-docking-system/skip_license_install.patch:10]: what's the diff here?

File: ports/qt-advanced-docking-system/skip_license_install.patch [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+ )
+-install(FILES 
+-    "${CMAKE_CURRENT_SOURCE_DIR}/LICENSE"
+-    "${CMAKE_CURRENT_SOURCE_DIR}/gnu-lgpl-v2.1.md"
+-    DESTINATION license 
+-    COMPONENT license
+-)
+ install(TARGETS qtadvanceddocking
+-    EXPORT qt-advanced-docking-systemConfig
++    EXPORT qt-advanced-docking-systemConfig
```
Comment (strega-nil) [inline ports/qt-advanced-docking-system/skip_license_install.patch:26]: what's the diff here?

File: ports/qt-advanced-docking-system/skip_license_install.patch [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+     LIBRARY DESTINATION lib
+     ARCHIVE DESTINATION lib
+@@ -100,8 +94,8 @@ write_basic_package_version_file("adsConfigVersion.cmake"
+     VERSION ${ads_VERSION}
+     COMPATIBILITY SameMajorVersion
+ )
+-
+-install(EXPORT qt-advanced-docking-systemConfig NAMESPACE qt-advanced-docking-system:: DESTINATION share/qt-advanced-docking-system)
++
++install(EXPORT qt-advanced-docking-systemConfig NAMESPACE qt-advanced-docking-system:: DESTINATION share/qt-advanced-docking-system)
```
Comment (strega-nil) [inline ports/qt-advanced-docking-system/skip_license_install.patch:37]: what's the diff here?

File: ports/qt-advanced-docking-system/skip_license_install.patch [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ 
+ target_include_directories(qtadvanceddocking PUBLIC 
+     "$<BUILD_INTERFACE:${ads_INCLUDE}>"
+@@ -119,4 +113,3 @@ if(BUILD_EXAMPLES)
+     add_subdirectory(example)
+     add_subdirectory(demo)
+ endif()
+-
```
Comment (strega-nil) [inline ports/qt-advanced-docking-system/skip_license_install.patch:45]: please remove this.

**Review by strega-nil — APPROVED** [review]
Looks good, just got some minor suggestions and questions :)

Also, what's going on with `analyze-test-results.ps1` and `generate-skip-list.ps1`? they have different permissions now, maybe? is there a reason for that?

File: ports/openblas/portfile.cmake (hunk 3 above)
Comment (BillyONeal) [reply ports/openblas/portfile.cmake:6]: Why is that?

File: ports/qt-advanced-docking-system/skip_license_install.patch (hunk 4 above)
Comment (BillyONeal) [reply ports/qt-advanced-docking-system/skip_license_install.patch:10]: Probably trailing whitespace VS Code automatically deleted when I saved the change :/

File: ports/qt-advanced-docking-system/skip_license_install.patch (hunk 5 above)
Comment (BillyONeal) [reply ports/qt-advanced-docking-system/skip_license_install.patch:26]: Ditto probably trailing whitespace

File: ports/qt-advanced-docking-system/skip_license_install.patch (hunk 6 above)
Comment (BillyONeal) [reply ports/qt-advanced-docking-system/skip_license_install.patch:37]: Ditto trailing whitespace

**Comment by BillyONeal** [comment]
> Also, what's going on with `analyze-test-results.ps1` and `generate-skip-list.ps1`? they have different permissions now, maybe? is there a reason for that?

`chmod +x` :)

File: scripts/azure-pipelines/windows/create-vmss.ps1 (hunk 2 above)
Comment (cenit) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: > @cbezault That 70GiB is for all 6 configurations

I sincerely doubt that. I have a 90.5 GB vcpkg folder for only a single configuration in windows-x64, and for sure I don't have not even a tenth of vcpkg ports installed....

Comment (cbezault) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: @cenit it's just for the archives. The downloads, buildtrees, packages and installed directories are what get really big.

Comment (cenit) [reply scripts/azure-pipelines/windows/create-vmss.ps1:297]: Understood. Thanks!

**Pull request merged**